File: frontend/types/sponsors.ts
import type { SourceAttribution } from "./common";

export interface SponsorRecordView {
  id: string;
  organisation_name: string;
  town_city: string | null;
  county: string | null;
  rating: string | null;
  routes: string[];
  source: SourceAttribution;
}

export interface SponsorSearchResponse {
  query: string;
  results: SponsorRecordView[];
  total: number;
  dataset_version: string | null;
  message: string | null;
  suggestions: SponsorRecordView[];
}

File: backend/app/schemas/sponsors.py
from __future__ import annotations

from pydantic import BaseModel, Field

from app.schemas.common import SourceAttribution


class SponsorRecordView(BaseModel):
    id: str
    organisation_name: str
    town_city: str | None = None
    county: str | None = None
    rating: str | None = None
    routes: list[str] = Field(default_factory=list)
    source: SourceAttribution


class SponsorSearchResponse(BaseModel):
    query: str
    results: list[SponsorRecordView]
    total: int
    dataset_version: str | None = None
    message: str | None = None
    suggestions: list[SponsorRecordView] = Field(default_factory=list)
